### DISCUSSION TITLE
[Monorepo] `tsconfig` & `project` parser configuration

### DISCUSSION THREAD

**Discussion by joris-fonck-loreal** [typescript-eslint/typescript-eslint#8728] [Technical Discussions]
### Body

Hello,

I'm trying to understand how the parser and lint works when we want to enable **Type Information** in a monorepo. Imagine a monorepo structure :

```
project/
├── packages/
│   ├── package-1/
│   │   ├── src/
│   │   │   └── ...
│   │   └── tsconfig.json
│   └── package-2/
│       ├── src/
│       │   └── ...
│       └── tsconfig.json
├── tsconfig.base.json
└── eslint.config.js
```

With `eslint.config.js` configuration:

```
export default {
    languageOptions: {
	parser: 'typescript-eslint',
	parserOptions: {
		sourceType: 'module',
		project: ['./tsconfig.base.json', './packages/package-1/tsconfig.json', './packages/package-2/tsconfig.json],
		tsconfigRootDir: process.cwd(),
	},
    },
    rules: {
        ...
    }
}
```

When linting, does ESLint use the closest tsconfig.json file to apply the rules? Or is it using the tsconfig.json file that includes the file? Both ?

Regards.

### Additional Info

Any additional info...

### Before you submit your discussion, please confirm the following. If any of these required steps are not taken, we may not be able to review your RFC. Help us to help you!

- [X] I have [read the discussions guidelines](https://typescript-eslint.io/contributing/discussions) and per those guidelines, this fits a discussion category, _not_ a help request or standard issue.
- [X] I have [searched for related discussions](https://github.com/typescript-eslint/typescript-eslint/discussions) and [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues) and found none that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

**Comment by bradzacher** [comment]
As per the discussions guidelines (which you ticked as having read) we do not use discussions as a support forum

https://typescript-eslint.io/contributing/discussions/

The discussion guideline contains alternatives for support.

**Reply by joris-fonck-loreal** [reply]
Sorry for that. But to me it's part of `Technical Discussions: Deeper questions about typescript-eslint's architecture and/or APIs` because it's about API. Will ask my question in other platform then.
